### PR TITLE
Fix operation on Windows

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
I updated the version of **minify** used for building my website and discovered it no longer worked on Windows, returning this error:

```
ERROR: CreateFile ./C:\dev\ccw\site: The filename, directory name, or volume label syntax is incorrect.
```

The folder `C:\dev\ccw\site` is the output location of my Jekyll based website. The `./` prefix that's been added is the source of the problem

This PR fixes the issues causing **minify** to fail on Windows:

* Uses the cross-platform package `path/filepath` instead of the Linux specific `path`
* Ensure directory detection is done using `os.PathSeparator`
* Compares file references in a platform specific way

I also added a little bit of context to errors, as a part of identifying where the original issue was happening.

Lastly, thanks for creating **minify** - it's greatly useful, and (literally orders of magnitude) faster than other tools I've auditioned.

### CODE REVIEW

**Comment by tdewolff** [comment]
Thanks for the complete PR including tests! And great that you've enjoyed using minify!

One thing that worries me is that the `fs.go` file is literally a copy of standard library functions (notably https://cs.opensource.google/go/go/+/refs/tags/go1.17.6:src/io/fs/walk.go and https://cs.opensource.google/go/go/+/refs/tags/go1.17.6:src/io/fs/fs.go). This is because an older version of Go does not include those files, thus for compatibility it was included. Apparently, the standard library works well for Windows somehow, which leads me to believe that something is wrong in how we use those functions in `main.go`. What do you think?

**Comment by theunrepentantgeek** [comment]
I suspect you're correct. Keeping the contents of `fs.go` consistent with the originals is useful for ongoing maintenance. 

I'll revisit my changes and will update this PR.

**Comment by theunrepentantgeek** [comment]
@tdewolff - I've updated the PR as discussed.

Key change was to modify the parameters to `DirFS()` (see `main.go`, line 631), and to treat absolute/relative paths differently inside the definition of `walkFn` (see `main.go`, line 587).

File: cmd/minify/main.go [hunk 1]
```diff
@@ -620,7 +628,7 @@ func createTasks(inputs []string, output string) ([]Task, []string, error) {
 				}
 				return nil
 			}
-			if err := WalkDir(DirFS("."), input, walkFn); err != nil {
+			if err := WalkDir(DirFS(input), ".", walkFn); err != nil {
```
Comment (tdewolff) [inline cmd/minify/main.go:631]: I think keeping this as it was will prevent the changes above for joining `root` with `path`. I don't think that the `walkFn` will ever have an absolute path?

**Review by tdewolff — COMMENTED** [review]
Added one comment which may be the result of my lack of understanding, the rest is looking very good! Thanks again!

File: cmd/minify/main.go (hunk 1 above)
Comment (theunrepentantgeek) [reply cmd/minify/main.go:631]: In my testing, `walkFn()` _was_ always called with a relative path - but I'm taking a conservative approach, concious that minify can be invoked in a number of different ways with different paramters.

I didn't want to introduce a bug by prepending the path with `root` if it wasn't needed, so I wrote the code defensively.

I can remove the `filepath.IsAbs()` test (line 588 of `main.go`) if you prefer.

**Comment by tdewolff** [comment]
Thanks for the PR!

File: cmd/minify/main.go (hunk 1 above)
Comment (tdewolff) [reply cmd/minify/main.go:631]: You were right, this was not working well with absolute input paths, thanks!

File: cmd/minify/main.go [hunk 2]
```diff
@@ -735,7 +747,13 @@ func minify(t Task) bool {
 	} else {
 		// rename original when overwriting
 		for i := range t.srcs {
-			if t.srcs[i] == t.dst {
+			sameFile, err := SameFile(t.srcs[i], t.dst)
+			if err != nil {
```
Comment (tdewolff) [inline cmd/minify/main.go:751]: Had to remove this part in case the output file doesn't exist. In case of an error we should continue execution

**Pull request merged**